projects/extension/build.py

```diff
@@ -90,16 +108,68 @@ def idempotent_sql_files() -> list[Path]:
     return paths
 
 
+def parse_feature_flag(path: Path) -> str | None:
+    with path.open(mode="rt", encoding="utf-8") as f:
+        line = f.readline()
+        if not line.startswith("--FEATURE-FLAG: "):
+            return None
+        ff = line.removeprefix("--FEATURE-FLAG: ").strip()
+        pattern = r"^[a-z_]+$"
+        if re.fullmatch(pattern, ff) is None:
+            print(
+                f"feature flag {ff} in {path.name} does not match the pattern {pattern}",
+                file=sys.stderr,
+            )
+            sys.exit(1)
+        return ff
+
+
+def check_sql_file_name(path: Path) -> None:
+    pattern = r"^\d\d\d-[a-z][a-z_-]*\.sql$"
+    if re.fullmatch(pattern, path.name) is None:
+        print(
+            f"{path} file name does not match the pattern {pattern}",
+            file=sys.stderr,
+        )
+        sys.exit(1)
+
+
+def file_number(path: Path) -> int:
+    return int(path.name[0:3])
+
+
 def check_idempotent_sql_files(paths: list[Path]) -> None:
```

Comment:
I suggest we add a comment mentioning the paths are sorted.
```suggestion
def check_idempotent_sql_files(paths: list[Path]) -> None:
```

Reply:
done